cloud/amazon/elastic_beanstalk.py

```diff
@@ -0,0 +1,467 @@
+            whether to create a new environment and swap CNAMEs
+    required: true
+    default: false
+extends_documentation_fragment: aws
+'''
+
+EXAMPLES = '''
+# Create an environment
+ - elastic_beanstalk:
+    application_name: my_application
+    region: us-east-1
+    application_version: 1.0
+    cname: my_cname
+    application_s3_bucket: my_bucket
+    application_s3_key: my_war_file
+    solution_stack: 64bit Amazon Linux 2015.03 v1.3.1 running Tomcat 8 Java 8
+    environment_options:
+      - Namespace: aws:autoscaling:launchconfiguration
+        OptionName: InstanceType
+        Value: t1.micro
+    state: present
+
+# Deploy a new application version to a new environment, and swap CNAMEs
+# with the existing environment
+ - elastic_beanstalk:
+   application_name: my_application
+   region: us-east-1
+   application_version: 2.0
+   cname: my_cname
+   application_s3_bucket: my_bucket
+   application_s3_key: my_war_file
+   solution_stack: 64bit Amazon Linux 2015.03 v1.3.1 running Tomcat 8 Java 8
+   environment_options:
+     - Namespace: aws:autoscaling:launchconfiguration
+       OptionName: InstanceType
+       Value: t1.micro
+   state: present
+   redeploy: true
+
+# Restart an environment
+  - elastic_beanstalk:
+      application_name: my_application
+      region: us-east-1
+      environment_name: my_environment
+      state: restarted
+
+# Terminate an environment
+  - elastic_beanstalk:
+    application_name: my_application
+    region: us-east-1
+    environment_name: my_environment
+    state: absent
+'''
+
+RETURN = '''
+Environment:
+  description: Details of the deployed environment
+  returned: success
+  type: list
+  sample:
+    ApplicationName: example-app
+    CNAME: example.us-west-2.elasticbeanstalk.com
+    EnvironmentName: example-env
+'''
+
+
+import string
+import time
+import random
+
+try:
+    import boto
+    import boto.beanstalk.exception
+except ImportError:
+    HAS_BOTO = False
+else:
+    HAS_BOTO = True
+
+def retrieve_environment(module, beanstalk):
+    """Return the dictionary that describes the environment from AWS, or None if not present"""
+
+    application_name = module.params.get("application_name")
+    environment_name = module.params.get("environment_name")
+
+    try:
+        environment_list = beanstalk.describe_environments(
+            application_name=application_name,
+            environment_names=[environment_name]
+        )["DescribeEnvironmentsResponse"]["DescribeEnvironmentsResult"]["Environments"]
+    except (boto.exception.BotoServerError, LookupError) as error:
+        module.fail_json(msg="Unable to retrieve environment information: '{0}'".format(error))
+    if len(environment_list) < 1:
+        environment = None
+    else:
+        environment = environment_list[0]
+        if environment["Status"] in ["Terminated"]:
+            environment = None
+    return environment
```

Comment:
Returning the environment as-supplied by AWS seems confusing, since the casing standard in AWS (CamelCase) doesn't match what's usually returned by Ansible modules (snake_case).